### ISSUE TITLE
 Improve user work summary at task panel 

### ISSUE THREAD

**Issue by thejoshwolfe** [Igalia/phpreport#341]
There's a box called User Work Summary that shows a "Week goal" among other things. How about putting some or all of the information from `viewWorkingHoursResultsReport.php` for you in that box? The goal would be greater awareness of acc hours and holiday hours.

**Comment by psaavedra** [comment]
The infomation that could be shown  (additionally to the current one):

* **Accumulated (Total) extra hours**: from 1-1-CURRENT_YEAR until TODAY (using the same logic implemented in `web/services/getExtraHoursReportService.php`)
* **Extra hours**: from Monday to Sunday of the current WEEK (using the same logic implemented in `web/services/getExtraHoursReportService.php`)
* **Workable hours**: from Monday to Sunday of the current WEEK (using the same logic implemented in `web/services/getExtraHoursReportService.php`)
* **Worked hours**: from Monday to Sunday of the current WEEK (using the same logic implemented in `web/services/getExtraHoursReportService.php`)
* **Pending holidays hours**: from Monday to Sunday of the current WEEK (using the same logic implemented in `web/services/getPendingHolidayHoursService.php`)

Maybe adding these new elements as part of the information provided in the `/web/services/getPersonalSummaryByDateService.php` action.

**Comment by jaragunde** [comment]
In the accumulated hours report, the field "total extra hours" means the total amount of extra hours since the beginning of the contract. I would not use this term for anything else, so I would rename this field:
> **Accumulated (Total) extra hours**: from 1-1-CURRENT_YEAR until TODAY (using the same logic implemented in `web/services/getExtraHoursReportService.php`)

We could call it "extra hours (year)". In general, I think it's a good idea to make the time period explicit in the field name: worked hours (week), extra hours (week), extra hours (year), etc.

**Comment by jaragunde** [comment]
There's a lot of data we could show, and the space is limited. Let's start with a user work summary that contains the following info:
* Worked hours (today)
* Worked hours (week)
* Goal (week)
* Extra hours (year)
* Pending holiday hours (year)
* Extra hours (total)

We would be removing the "month" item, which is the least useful.

**Comment by psaavedra** [comment]
* +1

**Comment by jaragunde** [comment]
Reviewed at #396:
```
commit ee5ef2f2f4c84d0a10859ed9f6b78c24ce77c9bf
Merge: 3364174a 861fce6b
Author: Jacobo Aragunde Pérez <jaragunde@igalia.com>
Date:   Fri Apr 27 13:56:23 2018 +0200

    [#341] Improve user work summary at task panel
    
    Include extra information and documentation.
    
    (Merge pull request #396 from Igalia/user-work-summary)

commit 861fce6b2708d6ab2529d463ce2b3c2af095f5ac
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Fri Apr 27 12:05:45 2018 +0200

    [#341] Added new fields and improved readability of the documentation

commit 84799626dd65093ac3f5a38de91ab7f621707914
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Fri Apr 27 11:47:59 2018 +0200

    [#341] Added a pointer to the User work summary help page

commit 3389e07baed25498d2a592ce38f7ad2fcd8cc5d7
Author: Jacobo Aragunde Pérez <jaragunde@igalia.com>
Date:   Thu Apr 26 17:25:43 2018 +0200

    [#341] Simplify CSS rules and text labels

commit 9ff8c9b4a24f5e3a217978dc279c221d6ac31c5e
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Thu Apr 26 16:19:13 2018 +0200

    [#341] Improve user work summary at task panel
    
     * Formated times from minutes to hours:minutes
     * Changed labels to comply with the ticket
     * Increased label size to add room for more text

commit d6155e1993ee18136cb0b54d25bedceb7ef6c87a
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Tue Apr 24 09:45:27 2018 +0200

    [#341] Improve user work summary at task panel
    
     * Imported UserFacade at getPersonalSummaryByDateService.php
     * Added holidays and extra hours to the service output
     * Printed information at tasks.js panel

commit 5dced1417a655fb74568ab9c5e392cc193c4060b
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Mon Apr 23 22:12:47 2018 +0200

    [#341] Improve user work summary at PHPreport task panel
    
        * Removed Monthly hours count

```

**Comment by jaragunde** [comment]
There are certain situations that produce wrong data. Must be investigated.

**Comment by jaragunde** [comment]
Reviewed at #397:
```
commit c2b4743e833fc3c707839b3266278b8ceb3f1111
Merge: ee5ef2f2 aff9fc98
Author: Jacobo Aragunde Pérez <jaragunde@igalia.com>
Date:   Tue May 1 10:57:17 2018 +0200

    [#341] Fix arithmetic operations on the user work summary
    
    (Merge pull request #397 from Igalia/user-work-summary-v2)

commit aff9fc980be04f23edc7aff3303f764c86ba56de
Author: Jacobo Aragunde Pérez <jaragunde@igalia.com>
Date:   Tue May 1 10:51:15 2018 +0200

    [#341] Fix vertical misalignment.

commit 87211cf1b21ac14dc1325a6a256092c08b72d76f
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Mon Apr 30 14:20:40 2018 +0200

    [#341] Reworked the arithmetic operations. Added corner case catch

```

**Comment by jaragunde** [comment]
Further improvements from gathered user feedback. Reviewed at #398:
```
commit a1968a7cd544ace2c6ee4e183259d68c6c5464ac
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Tue May 8 15:51:33 2018 +0200

    [#341] Changed field order. Display values in days, hours and minutes.
    
    * Field order: list extra hours (year) and (total) next to each other
    
    * Display extra hours and pending holidays in days, if values greater
      than one week
    
    * Extracted data calculations logic into a function
    
    (Squash pull request #398 from Igalia/user-work-summary-v3)

```

**Comment by pabelenda** [comment]
Seems like the calculations are still not well done.

For example, having a journey of 8 hours, under some circumstances, in the user work summary we could be having something like:

Pending holidays: `2 days 08:00 h`

or

Pending holidays: `2 days 07:59 h`

Instead of:

Pending holidays: `3 days 00:00 h`

I've changed the calculations approach. First step is to `round` and translate the input time into minutes and then do all the calculations. This way we do not loose precision.

PR will come soon.

**Comment by jaragunde** [comment]
Merged more improvements from PR #406:
```
commit b73281599c9fa8b93c045e043039e9a664b19a2d 
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Tue May 22 14:33:39 2018 +0200

    [#341] Refactored the calculations made at formatHours function

```

Hope this is the final patch!

**Comment by jaragunde** [comment]
Another commit with cosmetic changes:
```
commit 6b8c7a99006f2c02caf429cf6c5018bb3809be8c
Author: Pablo Abelenda <pabelenda@igalia.com>
Date:   Thu May 31 09:50:28 2018 +0200

    [#341] Improved readability of the user work summary
    
     * Added new elements as titles to group data in sections
     * Bolded titles, not data itself
     * Created a variable to apply style only to the titles

```
